torchtitan/experiments/flux/train.py

```diff
@@ -182,7 +182,8 @@ def train_step(self, input_dict: dict[str, torch.Tensor], labels: torch.Tensor):
             or self.step == self.job_config.training.steps
         ):
             model.eval()
-            self.eval_step()
```

Comment:
I suggest that we move `eval` to the outer loop and upstream to the core train.py. The core Trainer can define an empty `eval()` such that all other models are not affected. Flux model can then overwrite the eval(). 

This allows us to 
1. Provide a generic in-trainer eval interface for all models.
2. Overlap eval with checkpointing, which can greatly improve the checkpointing. performance

Reply:
Sure! I think it's a good suggestion here for in-train eval(). However, currently we only have flux use `eval_step()` function. To change other models , I will make the change in another diff